### DISCUSSION TITLE
use pre request script

### DISCUSSION THREAD

**Discussion by flocheke7** [Kong/insomnia#7519] [Help]
0

I'm trying to add a pre request script to my request in insomnia. Initially this script only makes a console.log

If I execute my request without this prerequest script, I have a good response

if I add this pre request script, I get this error

Timeout: Hidden browser window is not responding 

when I look in the console I see this error

hiddenWindowPort is undefined

I looked for how I could configure this port but I couldn't find any documentation on it.

how can I configure insomnia so that it executes this script?

thx

**Comment by ihexxa** [comment]
Hello @flocheke7 , theoretically you are able to run script after restarting the app.

It would be great if you could share some information which would help to identify the problem: 1) could you share the version you are using? 2) could you share steps for reproducing it or a minimal export file to reproduce it?

And thanks a lot for your feedback.

**Comment by flocheke7** [comment]
Hello Ihexxa 

II'm using the latest version which I downloaded yesterday morning

![image](https://github.com/Kong/insomnia/assets/16335253/2ebad0ed-863b-45f2-a9d5-e673356d313f)

for the rest service that I call, I use a local pring boot application which exposes a rest api 
so dna sl'utrl, I put: http://localhost:8080/x/v1/callForInasti 

in the script: 
console.log('ffffffffff' );



The first time I open insomnia, this call works and I see the logs in timeline. 

But then when I return this call, and the request remains pending, I don't know how to cancel it. I have to kill the insomnia spots in task manager. I don't see anything in timeline or in the logs



however on the backend side the rest service responded (I am logging their response)


Another question, for application needs, I need certain functionalities that I cannot find in the standard external libs of insomnia ex: forge 
Based on what I found on the internet, I do this:

insomnia.sendRequest("https://cdnjs.cloudflare.com/ajax/libs/dayjs/1.11.11/dayjs.min.js", (err, res) => {
 //  console.log('xxxx' + res.text());
 //  insomnia.globals.set("forgeJS", res.text());
  if (err) {
            console.log('rrrrrrrrr' + err.text());
        } else {
					 console.log('ghghghghghgh' + err.text());
            insomnia.baseEnvironment.set("forgeJS", res.text());
					//eval(insomnia.baseEnvironment.get("forgeJS"))
        }
})



I have a timeout for the get whereas if I type the same url in a browser or a normal query, I get a very fast response
(test with and without vpn)
(in insomnia config , i have not proxy)

how i can call this kind of lib in insomnia ?

thx for your help

**Comment by ihexxa** [comment]
This is weird, because if you've observed logs from the server side, which means that the script has already been successfully executed.
For the 'forgeJs" question, IIUC, currently Insomnia doesn't support importing external script from customer, probably evaluating forge.js took to much time so that it failed to execute the script.
